packages/algob/src/lib/deployer.ts

```diff
@@ -0,0 +1,34 @@
+import { txWriter, TxWriterImpl } from "../internal/tx-log-writer";
+import { mkAccountIndex } from "../lib/account";
+import { AlgoOperator } from "../lib/algo-operator";
+import { loadASAFile } from "../lib/asa";
+import type {
+  Accounts,
+  AlgobRuntimeEnv,
+  ASADefs,
+  CheckpointRepo,
+  DeployerConfig
+} from "../types";
+import { loadCheckpointsRecursive } from "./script-checkpoints";
+
+// intialize deployer config obj
+export class MkDeployerConfig implements DeployerConfig {
+  runtimeEnv: AlgobRuntimeEnv;
+  cpData: CheckpointRepo;
+  asaDefs: ASADefs;
+  algoOp: AlgoOperator;
+  txWriter: txWriter;
+  accounts: Accounts;
+
+  constructor (
+    runtimeEnv: AlgobRuntimeEnv,
+    algoOp: AlgoOperator
+  ) {
+    this.runtimeEnv = runtimeEnv;
+    this.cpData = loadCheckpointsRecursive();
+    this.algoOp = algoOp;
+    this.accounts = mkAccountIndex(runtimeEnv.network.config.accounts); ;
+    this.txWriter = new TxWriterImpl('');
+    this.asaDefs = loadASAFile(this.accounts);
+  }
+}
```

Comment:
this should be moved to `src/internal`.  The idea is following:

* everything in `src/lib` can be accessible to the user
* everything in `src/internal` shouldn't be presented to the user

I don't think you should be able to create a `DeployerConfig`.

Reply:
Ok, understood.

Reply:
Moving to `src/internal/util/deployer.ts`

Reply:
Please don't move it to `util`.
`util` should be reserved for small common things. In general we should avoid it in fact. we can have it in `src/internal/deployer_cfg.ts`

Reply:
Ok, noted.